Implement reset method in BlockDataCollector

mzstic opened the pull request:
Since Symfony 3.4 classes implementing DataCollectorInterface should implement reset() method.

I am targeting 3.x branch, because I'm only adding new method. No BC changes. There's no reason for waiting for new major release.

Closes https://github.com/sonata-project/SonataAdminBundle/issues/4699
Didn't know how else should I refer to issue in another bundle...

## Changelog

```markdown
### Added
- Implement reset method in `BlockDataCollector` to be compatible with Symfony 3.4
```
## Subject
Added one method to BlockDataCollector to implement new version of DataCollectorInterface correctly.

greg0ire commented:
Please move the subject out of the changelog. Also, the collector is not an interface, you cannot implement it. Please change the changelog message accordingly

mzstic commented:
@greg0ire I added unit test for BlockDataCollector. I'm just not sure ho style CI works... I understand there are some code style errors in files I changed, but style CI reports some errors in files I haven't touched. I'm just not sure if I should fix all of it or just the files I actually changed...?

greg0ire commented:
Either wait for us to fix those new issues or do it in a separate commit. These come from a new fixer in php-cs-fixer.

OskarStark commented on Tests/Profiler/DataCollector/BlockDataCollectorTest.php:
```diff
@@ -0,0 +1,84 @@
+        $this->assertEquals($expectedRealBlocks, $blockDataCollector->getRealBlocks());
+
+        $blockDataCollector->reset();
+        $this->assertEquals([], $blockDataCollector->getEvents());
```
assertEmpty would be better IMO

Soullivaneuh commented on Tests/Profiler/DataCollector/BlockDataCollectorTest.php:
```diff
@@ -0,0 +1,84 @@
+use PHPUnit\Framework\TestCase;
+use Sonata\BlockBundle\Profiler\DataCollector\BlockDataCollector;
+
+class BlockDataCollectorTest extends TestCase
```
* `final`
* Please add your author tag.

greg0ire commented:
Ping @mzstic !

OskarStark commented on Tests/Profiler/DataCollector/BlockDataCollectorTest.php:
```diff
@@ -0,0 +1,84 @@
+use PHPUnit\Framework\TestCase;
+use Sonata\BlockBundle\Profiler\DataCollector\BlockDataCollector;
+
+final class BlockDataCollectorTest extends TestCase
```
please add `@author` tags to the newly created classes

OskarStark commented:
I updated your changelog @mzstic and added 2 commits to your PR, so please pull this feature branch before force push.

Please rebase this, this should make StyleCI happy 😄

SonataCI commented:
Could you please rebase your PR and fix merge conflicts?

jordisala1991 commented:
fixed in https://github.com/sonata-project/SonataBlockBundle/pull/439
